src/MonitoringFunctions.Common/InstallScriptRunner.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Kusto.Cloud.Platform.IO;
+using MonitoringFunctions.Models;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+
+namespace MonitoringFunctions
+{
+    /// <summary>
+    /// This type contains the methods to run install scripts with specified parameters and parse the results.
+    /// </summary>
+    internal static class InstallScriptRunner
+    {
+        /// <summary>
+        /// Executes the install script with given arguments.
+        /// </summary>
+        /// <param name="commandLineArgs">Arguments to be passed to the script.</param>
+        /// <returns>Returns a <see cref="ScriptExecutionResult"/> object containing the output of the execution.</returns>
+        /// <remarks>The script to be run depends on the platform. On windows, powershell script is run.
+        /// On other operating systems, bash script is used.</remarks>
+        public static async Task<ScriptExecutionResult> ExecuteInstallScriptAsync(string? commandLineArgs)
+        {
+            ProcessStartInfo processStartInfo = GetScriptProcessStartInfo(out string scriptName, commandLineArgs);
+
+            processStartInfo.CreateNoWindow = true;
+            processStartInfo.UseShellExecute = false;
+            processStartInfo.RedirectStandardOutput = true;
+            processStartInfo.RedirectStandardError = true;
+
+            Process installScriptProc = Process.Start(processStartInfo);
+            string consoleOutput = await installScriptProc.StandardOutput.ReadToEndAsync().ConfigureAwait(false);
+            string consoleError = await installScriptProc.StandardError.ReadToEndAsync().ConfigureAwait(false);
+
+            return new ScriptExecutionResult()
+            {
+                ScriptName = scriptName,
+                Output = consoleOutput,
+                Error = consoleError
+            };
+        }
+
+        /// <summary>
+        /// Parses the output of the script when executed in DryRun mode and finds out Primary and Legacy urls.
+        /// </summary>
+        /// <param name="output">Output of the script execution in DryRun mode</param>
+        /// <returns>Object containing primary and legacy runs</returns>
+        public static ScriptDryRunResult ParseDryRunOutput(string? output)
+        {
+            string primaryUrlIdentifier = "Primary named payload URL: ";
+            string legacyUrlIdentifier = "Legacy named payload URL: ";
+
+            ScriptDryRunResult result = new ScriptDryRunResult();
+
+            using StringStream stringStream = new StringStream(output);
+            using StreamReader streamReader = new StreamReader(stringStream);
```

Comment:
nit: I see it's pre-existing code but I'm curious why it's not using `StringReader` to parse the string.

```suggestion
            using StringReader streamReader = new StringReader(output);
```

Reply:
`StringReader` only has a constructor that takes a string. That means we need to have the whole console output as a single string in memory. Not a huge issue, but I'm not sure if it is worth changing.

Reply:
We already have the entire output as a single string - passed in the `output` parameter. Hope I'm not reading it wrong.

Reply:
Ouch, you are right. I didn't look carefully, thought `output` was a stream. I merged this already but I will get that in the next PR.